src/collectives/device/all_to_all.h

```diff
@@ -18,18 +18,24 @@ class ncclFunction<ncclFuncAllToAll, ALGO, PROTO, FUNC, T, UNROLL> {
       const int nChannels = args->coll.nChannels;
       struct ncclDevComm* comm = args->comm;
       struct ncclChannel* channel = comm->channels+blockIdx.x;
-      struct ncclRing* ring = &channel->ring;
+      struct scklGraph* graph = &channel->graph;
       const int stepSize = comm->buffSizes[NCCL_PROTO_SIMPLE] / (sizeof(T)*NCCL_STEPS);
       const int chunkSize = stepSize * ALLTOALL_CHUNKSTEPS;
       const int nranks = comm->nRanks;
       const ssize_t loopSize = nChannels*(ssize_t)chunkSize;
       const ssize_t size = args->coll.count;
 
-
       // Compute pointers
       const T * __restrict__ thisInput = (const T*)args->sendbuff;
       T * __restrict__ thisOutput = (T*)args->recvbuff;
-      printf("This is an empty function! %d %d %d %d\n", (int) size, tid, bid, (int) sizeof(T));
+      ncclPrimitives<UNROLL, ALLGATHER_CHUNKSTEPS/ALLGATHER_SLICESTEPS, ALLGATHER_SLICESTEPS, T, 3, 3, 1, FUNC>
+        prims(tid, nthreads, graph->recv, graph->send, thisOutput, stepSize, channel, comm, ncclShmem->ptrs, 0);
+      if (tid == 0 && bid == 0){
+        printf("connected to %d %d %d\n", graph->send[0], graph->send[1], graph->send[2]);
+      }
+      int testSize = min(chunkSize, (int)size/nChannels/nranks);
+      prims.directSend(thisInput, 0, testSize);
+      prims.directRecv(thisOutput, 0, testSize);
```

Comment:
This is not all to all?

Reply:
Nope, just an empty alltoall to make sure the connections work.